roles/openshift_logging_curator/tasks/main.yaml

```diff
@@ -92,36 +72,136 @@
-    curator_cpu_request: "{{ openshift_logging_curator_cpu_request | min_cpu(openshift_logging_curator_cpu_limit | default(none)) }}"
-    curator_memory_limit: "{{ openshift_logging_curator_memory_limit }}"
-    curator_replicas: "{{ openshift_logging_curator_replicas | default (1) }}"
-    curator_node_selector: "{{openshift_logging_curator_nodeselector | default({})}}"
-  check_mode: no
-  changed_when: no
-
-- name: Set Curator DC
-  oc_obj:
-    state: present
-    name: "{{ curator_name }}"
-    namespace: "{{ openshift_logging_namespace }}"
-    kind: dc
-    files:
-      - "{{ tempdir }}/templates/curator-dc.yaml"
-    delete_after: true
+# Deployment Config - v2.x
+- when: not openshift_logging_es5_techpreview
+  block:
+    - name: Generate Curator deploymentconfig
+      template:
+        src: "{{ __base_file_dir }}/curator.j2"
+        dest: "{{ tempdir }}/templates/curator-dc.yaml"
+      vars:
+        component: "{{ curator_component }}"
+        logging_component: curator
+        deploy_name: "{{ curator_name }}"
+        image: "{{openshift_logging_curator_image_prefix}}logging-curator:{{openshift_logging_curator_image_version}}"
+        es_host: "{{ openshift_logging_curator_es_host }}"
+        es_port: "{{ openshift_logging_curator_es_port }}"
+        curator_cpu_limit: "{{ openshift_logging_curator_cpu_limit }}"
+        curator_cpu_request: "{{ openshift_logging_curator_cpu_request | min_cpu(openshift_logging_curator_cpu_limit | default(none)) }}"
+        curator_memory_limit: "{{ openshift_logging_curator_memory_limit }}"
+        curator_replicas: "{{ openshift_logging_curator_replicas | default (1) }}"
+        curator_node_selector: "{{openshift_logging_curator_nodeselector | default({})}}"
+      check_mode: no
+      changed_when: no
+
+    - copy:
+        src: "{{ __base_file_dir }}/curator.yml"
+        dest: "{{ tempdir }}/curator.yml"
+      changed_when: no
+
+    - import_role:
+        name: openshift_logging
+        tasks_from: patch_configmap_files.yaml
+      vars:
+        configmap_name: "logging-curator"
+        configmap_namespace: "logging"
+        configmap_file_names:
+          - current_file: "config.yaml"
+            new_file: "{{ tempdir }}/curator.yml"
+
+    - name: Set Curator configmap
+      oc_configmap:
+        state: present
+        name: "logging-curator"
+        namespace: "{{ openshift_logging_namespace }}"
+        from_file:
+          config.yaml: "{{ tempdir }}/curator.yml"
+
+    - name: Set Curator DC
+      oc_obj:
+        state: present
+        name: "{{ curator_name }}"
+        namespace: "{{ openshift_logging_namespace }}"
+        kind: dc
+        files:
+        - "{{ tempdir }}/templates/curator-dc.yaml"
+        delete_after: true
+
+# Cron Job - v5.x
+- when: openshift_logging_es5_techpreview | bool
+  block:
+    # Keep the old DC around
+    - name: Scale the old DC to 0
+      oc_scale:
+        name: "{{ curator_name }}"
```

Comment:
@ewolinetz there is a potential risk that $curator_name will have different value during the upgrade.
Am I being paranoiac?

Reply:
It shouldn't change during the upgrade... we derive it here:
https://github.com/openshift/openshift-ansible/pull/7047/files#diff-f4c6ebdf7da2e4a833073d1f89e08221R72